### ISSUE TITLE
django.shortcuts.redirect should use a type Literal overload for return value

### ISSUE THREAD

**Issue by silviogutierrez** [typeddjango/django-stubs#378]
# Bug report

## What's wrong
As of a recent `django-stubs` release, and `HttpResponseRedirect` and `HttpResponsePermanentRedirect` are no longer compatible with each other (rightly so, they only share an ancestor but the latter does not inherit from the former).

This means to annotate a view's response, you need to use something like `Union[TemplateResponse, HttpResponseRedirect, HttpResponsePermanentRedirect]` for your typical https://en.wikipedia.org/wiki/Post/Redirect/Get . Obviously you could just annotate with `HttpResponse` but I would argue that's too weak.

## How is that should be

Calling redirect with `permanent=True` should return one type, and `permanent=False` return another. This is possible with `Literal` types now, see https://mypy.readthedocs.io/en/stable/literal_types.html.

Happy to take a stab at this assuming we're ok with using newer MyPy features like that above. I'm not quite sure if we should import Literal from `typing` or from somewhere else if we support lower Python versions. Any guidance on these imports would be great.

Thanks for all the work on this library. It's really improved my Mypy experience.

## System information

- OS:
- `python` version: 3.8
- `django` version: 3.0
- `mypy` version: 0.770
- `django-stubs` version: 1.5.0

**Comment by kszmigiel** [comment]
Hello @silviogutierrez and thanks for report!

Are you able to precisely point out django-stubs version in which this bug hadn't exist? It would help me a lot solving this.

**Comment by silviogutierrez** [comment]
@kszmigiel : sure thing.

1.2 had `redirect` returning `HttpResponseRedirect` *only*. In reality, this was incorrect as we know it can return `HttpResponsePermanentRedirect` too. See here:

https://github.com/typeddjango/django-stubs/blob/v1.2.0/django-stubs/shortcuts.pyi

1.3 was updated to return a `Union` of both types. Link here:

https://github.com/typeddjango/django-stubs/blob/v1.3.0/django-stubs/shortcuts.pyi

To be clear, it's not a bug per se. I think the issue just defaulted to bug. This is really more of an improvement to `redirect`. Apologies if I messed it up on my end.
